src/waveform/waveformwidgetfactory.cpp

```diff
@@ -530,7 +530,7 @@ bool WaveformWidgetFactory::setWidgetTypeFromHandle(int handleIndex, bool force)
         // it since the size didn't change.
         //viewer->resize(viewer->size());
         widget->resize(viewer->width(), viewer->height());
-        widget->setTrack(pTrack);
+        //         widget->setTrack(pTrack);
```

Comment:
QGLWidget::initializeGL needs to be called on `widget->getWidget()` before `widget->setTrack(pTrack)`. The call to `widget->resize` above causes Qt to call QGLWidget::initializeGL, however, I could not figure out how to make Qt process that signal before the call to `widget->setTrack(pTrack)` here. I tried `QApplication::QApplication::sendPostedEvents(widget->getWidget())` and `QApplication::processEvents` but neither worked. Does anyone have ideas how to solve this?